donkeycar/templates/cv_control.py

```diff
@@ -158,48 +47,174 @@ def drive(cfg, args):
+    #
+    # There are 5 programmable webui buttons, "web/w1" to "web/w5"
+    # adding a button handler for a webui button
+    # is just adding a part with a run_condition set to
+    # the button's name, so it runs when button is pressed.
+    #
+    have_joystick = ctr is not None and isinstance(ctr, JoystickController)
+
+    # button to toggle recording
+    if cfg.TOGGLE_RECORDING_BTN:
+        print(f"Toggle recording button is {cfg.TOGGLE_RECORDING_BTN}")
+        if cfg.TOGGLE_RECORDING_BTN.startswith("web/w"):
+            V.add(Lambda(lambda: recording_control.toggle_recording()), run_condition=cfg.TOGGLE_RECORDING_BTN)
+        elif have_joystick:
+            ctr.set_button_down_trigger(cfg.TOGGLE_RECORDING_BTN, recording_control.toggle_recording)
+
+    # Buttons to tune PID constants
+    if cfg.DEC_PID_P_BTN and cfg.PID_P_DELTA:
+        print(f"Decrement PID P button is {cfg.DEC_PID_P_BTN}")
+        if cfg.DEC_PID_P_BTN.startswith("web/w"):
+            V.add(Lambda(lambda: dec_pid_p()), run_condition=cfg.DEC_PID_P_BTN)
+        elif have_joystick:
+            ctr.set_button_down_trigger(cfg.DEC_PID_P_BTN, dec_pid_p)
+    if cfg.INC_PID_P_BTN and cfg.PID_P_DELTA:
+        print(f"Increment PID P button is {cfg.INC_PID_P_BTN}")
+        if cfg.INC_PID_P_BTN.startswith("web/w"):
+            V.add(Lambda(lambda: inc_pid_p()), run_condition=cfg.INC_PID_P_BTN)
+        elif have_joystick:
+            ctr.set_button_down_trigger(cfg.INC_PID_P_BTN, inc_pid_p)
+    if cfg.DEC_PID_D_BTN and cfg.PID_D_DELTA:
+        print(f"Decrement PID D button is {cfg.DEC_PID_D_BTN}")
+        if cfg.DEC_PID_D_BTN.startswith("web/w"):
+            V.add(Lambda(lambda: dec_pid_d()), run_condition=cfg.DEC_PID_D_BTN)
+        elif have_joystick:
+            ctr.set_button_down_trigger(cfg.DEC_PID_D_BTN, dec_pid_d)
+    if cfg.INC_PID_D_BTN and cfg.PID_D_DELTA:
+        print(f"Increment PID D button is {cfg.INC_PID_D_BTN}")
+        if cfg.INC_PID_D_BTN.startswith("web/w"):
+            V.add(Lambda(lambda: inc_pid_d()), run_condition=cfg.INC_PID_D_BTN)
+        elif have_joystick:
+            ctr.set_button_down_trigger(cfg.INC_PID_D_BTN, inc_pid_d)
+
+    #
+    # Choose what inputs should change the car.
+    #
+    # TODO: when we merge pose estimate branch, update 'angle' to 'steering'
+    class DriveMode:
+        def run(self, mode,
+                    user_angle, user_throttle,
+                    pilot_angle, pilot_throttle):
+            if mode == 'user':
+                return user_angle, user_throttle
+
+            elif mode == 'local_angle':
+                return pilot_angle if pilot_angle else 0.0, user_throttle
 
+            else:
+                return pilot_angle if pilot_angle else 0.0, \
+                       pilot_throttle * cfg.AI_THROTTLE_MULT \
+                           if pilot_throttle else 0.0
+
+    V.add(DriveMode(),
+          inputs=['user/mode', 'user/angle', 'user/throttle',
```

Comment:
`user/steering` rather?

Reply:
The version of add_user_control() in this branch outputs 'user/angle', so we need to do that here.